temporalcli/commands.go

```diff
@@ -495,6 +495,8 @@ func (c *TemporalCommand) preRun(cctx *CommandContext) error {
 		}
 	}
 	cctx.JSONShorthandPayloads = !c.NoJsonShorthandPayloads
+	cctx.Context, _ = context.WithDeadline(cctx.Context, time.Now().Add(c.CommandTimeout.Duration()))
```

Comment:
```suggestion
	if c.CommandTimeout.Duration() > 0 {
		cctx.Context, _ = context.WithTimeoutCause(
			cctx.Context,
			c.CommandTimeout.Duration(),
			fmt.Errorf("command timed out"),
		)
	}
```

I think this or something like it makes more sense than using deadline and always setting it even if not set by user.